File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> query;
        for (int i : nums){
            if(query.find(i)!=query.end()){
                return true;
            }
            query.insert(i);
        }
        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # As I iterate I will check if element exists behind me if not add elemet and move on
        query = set()
        for i in nums:
            if i in query:
                return True
            query.add(i)
        return False